### PR TITLE
fix: Implement validation for request parameter idScheme in tracker controller [DHIS2-13324]

### PR DESCRIPTION
**Pull request by enricocolasante**
This PR introduce a way to validate request parameters in `TrackerImportController`

At the moment only `idScheme` parameter is validated but others can be easily added.

Next steps would be:

- Add all the request parameter to validation
- Extract `validateEnum` method to a more generic location so it can be used by other controllers
- Find an elegant way to present more than one validation error at a time

### CODE REVIEW

**Review by teleivo — COMMENTED** [review]
Great you suggest a solution fixing this entire category of issues with our query parameter values 😋 I added some small suggestions.

Something we can discuss later but is that I think it would be beneficial to make this validation independent of our importer. Any of our endpoints benefit from this validation. Meaning I think it could be as generic as dealing with validating on the HTTP level. Since you call this in every controller method this smells like a job for a middleware. So not sure what the Spring equivalent of a middleware is. Servlet do already provide this I think 🤔 Validating the HTTPServletRequest for example.

File: dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/exception/InvalidEnumValueException.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+@Getter
+public class InvalidEnumValueException extends Exception
+{
+
+    private final String wrongValue;
+
+    private final String field;
+
+    private final List<String> allowedValues;
```
[the remark this answers is not in the document]
Comment (teleivo) [reply dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/exception/InvalidEnumValueException.java:42]: Sounds good!

File: dhis-2/dhis-web-api-test/src/test/java/org/hisp/dhis/webapi/controller/TrackerImportControllerTest.java [hunk 2]
```diff
@@ -41,6 +41,21 @@
 class TrackerImportControllerTest extends DhisControllerConvenienceTest
 {
 
+    @Test
+    void testSyncPostJsonTrackerInvalidReportMode()
+    {
+        assertWebMessage( "Bad Request", 400, "ERROR", "400 Value INVALID is not a valid report mode",
```
[the remark this answers is not in the document]
Comment (enricocolasante) [reply dhis-2/dhis-web-api-test/src/test/java/org/hisp/dhis/webapi/controller/TrackerImportControllerTest.java:47]: Lets talk about this.
I think this PR deserve a follow up PR to validate all the parameters for tracker importer, it shouldn't be a lot of work now

**Comment by enricocolasante** [comment]
> Great you suggest a solution fixing this entire category of issues with our query parameter values yum I added some small suggestions.
> 
> Something we can discuss later but is that I think it would be beneficial to make this validation independent of our importer. Any of our endpoints benefit from this validation. Meaning I think it could be as generic as dealing with validating on the HTTP level. Since you call this in every controller method this smells like a job for a middleware. So not sure what the Spring equivalent of a middleware is. Servlet do already provide this I think thinking Validating the HTTPServletRequest for example.

I absolutely agree.
The idea of this PR was to explore our needs and understand how to proceed towards a common way to validate request parameters through all the endpoints

**Review by lucaCambi77 — APPROVED** [review]

**Pull request merged**